### PR TITLE
REP-3906 Keystone v2 Tenanted Roles Handling

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

File: repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala [hunk 1]
```diff
@@ -310,46 +309,55 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
...
+              role.tenantId.forall(roleTenantId =>
+                tenantFromUri.exists(roleTenantId.equals)))
+          case _ =>
+            roles
+        }
+      }
+
+      def isUserPreAuthed(roles: Seq[Role]): Boolean = {
+        Option(config.getPreAuthorizedRoles) exists { preAuthedRoles =>
+          preAuthedRoles.getRole.asScala.intersect(roles.map(_.name)).nonEmpty
```
Comment (wdschei) [inline repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala]: This is why I like Scala so much. So simple and elegant while staying readable.

File: repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala [hunk 2]
```diff
@@ -718,6 +720,9 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
     override def onLifecycleEvent(event: LifecycleEvents): Unit = {
       logger.debug(s"Received Lifecycle Event: $event")
     }
+
+    case class RegisteredFeed(id: String, unique: String)
```
Comment (wdschei) [inline repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala]: Was this IntelliJ shenanigans or intentional?

Comment (dmnjohns) [inline repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala]: IntelliJ shenanigans. It likes inner classes to be declared at the bottom. It doesn't bother me at all, so I let it have its way.

Comment (wdschei) [inline repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala]: Doesn't bother me either, just making sure.

**Comment by dmnjohns** [comment]
retest this please

**Comment by dmnjohns** [comment]
I really don't know why those BodyPatcherTests are failing. They are throwing `NullPointerException` and `IllegalArgumentException` from the test class, which is just strange. Oh, and I can't reproduce that behavior locally.

**Comment by wdschei** [comment]
Are they currently passing in the `repose-8-staging` branch?

**Comment by dmnjohns** [comment]
They are locally. However, they appear to be failing on Jenkins based on:
https://jenkins-proto.openrepose.org/job/repose-gradle-pull_request/368/

I kicked off another build on Jenkins to make certain:
https://jenkins-proto.openrepose.org/job/repose-gradle-pull_request/372/

**Comment by wdschei** [comment]
So if the 372 run is good, then this is good?
Or if the 372 run is good, then it proves something else is broke?

**Comment by wdschei** [comment]
Either way, the 372 run passed.

**Comment by wdschei** [comment]
That brought it down to only a single failure.

**Comment by dmnjohns** [comment]
Weird. For that test, it looks like the handling isn't being created. From the logs, it appears that there is a failure in applying the patch to the request.

> 2016-05-19 00:04:36,774 16078 [qtp1926744025-52] ERROR  org.openrepose.powerfilter.PowerFilterChain - Failure in filter: BodyPatcherFilter  -  Reason: element bar does not exist at path /
gnieh.diffson.PatchException: element bar does not exist at path /

Looking at the test, though, it appears that the request body should contain a JSON object with the "bar" element. Also, the config looks correct, so I'm not sure why this is failing.

**Comment by adrianjgeorge** [comment]
These are being flaky on the staging branch too, so they are in no way an impediment to this being merged

**Comment by wdschei** [comment]
Cool. Everything else looked good. I'll merge now.

**Pull request merged**